src/main/kotlin/no/nav/syfo/behandler/api/BehandlerApi.kt

```diff
@@ -45,14 +48,28 @@ fun Route.registerBehandlerApi(
             }
         }
         get(search) {
-            val token = getBearerHeader()
-                ?: throw IllegalArgumentException("No Authorization header supplied")
-            val search = this.call.request.headers["searchstring"]
-                ?: throw IllegalArgumentException("No searchstring supplied")
-            val behandlere = behandlerService.searchBehandlere(
-                searchStrings = search,
-            )
-            call.respond(behandlere.toBehandlerDTOListUtenRelasjonstype())
+            withValidToken {
+                val search = this.call.request.headers["searchstring"]
+                    ?: throw IllegalArgumentException("No searchstring supplied")
+                val behandlere = behandlerService.searchBehandlere(
+                    searchStrings = search,
+                )
+                call.respond(behandlere.toBehandlerDTOListUtenRelasjonstype())
+            }
+        }
+        get(behandlerRef) {
+            withValidToken {
+                val behandlerRef: String = this.call.request.headers["behandlerRef"]
+                    ?: throw IllegalArgumentException("No behandlerRef supplied")
+                val behandler = behandlerService.getBehandler(
+                    behandlerRef = UUID.fromString(behandlerRef)
+                )
+                if (behandler != null) {
+                    call.respond(behandler.toBehandlerDTO(behandlerType = null))
```

Comment:
Satte `behandlerType` til `null`, men er det ønskelig å få med typen her også?

Reply:
Spørs om vi trenger å vise den fram kanskje? Kan sikkert vente med det til ny PR (ser ut som det vil kreve spørringer i flere tabeller)

Reply:
Jeg er ikke sikker på om denne typen kan settes uten at det er i kontekst av en arbeidstaker (kan feks være FASTLEGE for en arbeidstaker og SYKMELDER for en annen).